core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -466,6 +466,13 @@ public void unparseCall(SqlWriter writer, SqlCall call, int leftPrec,
     }
   }
 
+  public void unparseBoolLiteral(SqlWriter writer,
+      SqlLiteral literal, int leftPrec, int rightPrec) {
+    Object value = literal.getValue();
+    writer.keyword(
+        !(value instanceof Boolean) ? "UNKNOWN" : (Boolean) value ? "TRUE" : "FALSE");
```

Comment:
Why do we use `value instanceof Boolean`, not the `value == null`？CASE condition has done the same thing, so I think `value == null` is clearer to me.